### DISCUSSION TITLE
please advise me the usage of Quectel as UE to test the network

### DISCUSSION THREAD

**Discussion by initChat** [srsran/srsRAN_Project#537] [General Help] [answered]
I have read in this discussion forum that many people use Quectel RM510 as UE for testing 4G/5G networks. I have started to use it and mostly test on Ping. Either,
```
>> ping -I wwan0 8.8.8.8
```
or ping with AT command in minicom
```
>> sudo minicom -s
>> AT+COPS?
>> AT+QPING=1,"8.8.8.8"
```
However, I wonder if Quectel could test more than ping. I would like to ask what everyone uses Quectel for in your 4G/5G network. 
Thank you very much.

This is my current setup
![quectel-quectel_rpi (2)](https://github.com/srsran/srsRAN_Project/assets/40858099/bc4a2a1a-4f75-48bd-8826-c73fbbbed055)

**Comment by pgawlowicz** [comment]
@herlesupreeth could you comment here?

**Comment by herlesupreeth** [comment]
@pchat-imm please find the way in which I use Quectel Modem

https://github.com/srsran/srsRAN_Project/discussions/426#discussioncomment-8233829

In addition to that you can also use iperf with `-B` option which allows you bind to an IP/interface to send traffic.

**Reply by herlesupreeth** [reply, marked as answer]
For example: Assuming Quectel modem interface (e.g. wwan0) has been assigned IP address 192.168.100.7. And, the IP address of the UPF interface (e.g. ogstun) is 192.168.100.1.

iperf server at UE

`sudo iperf -u -s -B 192.168.100.7`

iperf client at UE

`sudo iperf -u -c 192.168.100.1 -t 50 -i 1 -b 150M -B 192.168.100.7`
